### PR TITLE
Added hashing

### PR DESCRIPTION
**Pull request by fontanellag**
**Issue(s):**  69

### Description

Added hashing.md page inside the Cairo Cheatsheet section which contains a concise description on how to perform a Pedersen and Poseidon hashes. Also added the related code to the listing folder

### CODE REVIEW

File: src/ch00/cairo_cheatsheet/hashing.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Hashing
+
+The hashing process is a deterministic transformation that allows converting input data of any length into a fixed-size value referred as "hash". There are a plenty of hashing algorithms: in Cairo we have Pedersen and Poseidon hashes. Pedersen hashes were used in the past (but still used in some scenario for backward compatibility) while Poseidon hashes are the standard nowadays since they were designed to be very efficient for Zero Knowledge proof systems. More details [here](https://book.cairo-lang.org/ch11-03-hash.html#hash-functions-in-cairo).
+
+In Cairo it's possibile to hash all the types that can be converted to `felt252` since they implement natively the `Hash` trait. It's also possible to hash more complex types like structs by deriving the Hash trait with the attribute `#[derive(Hash)]` but only if all the struct's fields are themselves hashable.
```
Comment (julio4) [inline src/ch00/cairo_cheatsheet/hashing.md:5]: possible typo

**Review by julio4 — COMMENTED** [review]
Nice example and explanation! Could you wrap the hashing_example in an mdbook anchor and use #rustdoc_include to hide tests in the book?

File: src/ch00/cairo_cheatsheet/hashing.md (hunk 1 above)
Comment (fontanellag) [reply src/ch00/cairo_cheatsheet/hashing.md:5]: Sorry where's the typo? Thanks

Comment (fontanellag) [reply src/ch00/cairo_cheatsheet/hashing.md:5]: Caught

**Comment by fontanellag** [comment]
Changes done!

**Review by julio4 — COMMENTED** [review]
Some proposed changes from this PR were moved to #161

**Pull request closed without merging**